src/main/java/org/primefaces/component/cascadeselect/CascadeSelectRenderer.java

```diff
@@ -116,9 +118,18 @@ protected void encodeInput(FacesContext context, CascadeSelect cascadeSelect, St
         writer.endElement("div");
     }
 
-    protected void encodeLabel(FacesContext context, CascadeSelect cascadeSelect, String valueToRender) throws IOException {
+    protected void encodeLabel(FacesContext context, CascadeSelect cascadeSelect, List<SelectItem> itemList, String valueToRender) throws IOException {
         ResponseWriter writer = context.getResponseWriter();
-        String placeholder = LangUtils.isNotBlank(valueToRender) ? valueToRender : cascadeSelect.getPlaceholder();
+
+        String itemLabel = valueToRender;
+        if (valueToRender != null) {
```

Comment:
why still the null check?

Reply:
You are right this is unnecessary now.  Removed.